src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisResultService.java

```diff
@@ -0,0 +1,152 @@
+            LoadFlowResult.ComponentResult.Status.valueOf(securityAnalysisResult.get().getPreContingencyStatus()),
+            new LimitViolationsResult(preContingencyLimitViolations),
+            new NetworkResult(Collections.emptyList(), Collections.emptyList(), Collections.emptyList())
+        );
+    }
+
+    @Transactional(readOnly = true)
+    public List<ContingencyToSubjectLimitViolationDTO> findNmKContingenciesResult(UUID resultUuid) {
+        assertResultExists(resultUuid);
+        List<ContingencyEntity> contingencies = contingencyRepository.findByResultIdAndStatusOrderByContingencyId(resultUuid, LoadFlowResult.ComponentResult.Status.CONVERGED.name());
+        return contingencies.stream().map(contingency -> {
+            List<SubjectLimitViolationFromContingencyDTO> subjectLimitViolations = contingency.getContingencyLimitViolations().stream()
+                .map(SubjectLimitViolationFromContingencyDTO::toDto)
+                .toList();
+            return new ContingencyToSubjectLimitViolationDTO(
+                contingency.getContingencyId(),
+                contingency.getStatus(),
+                contingency.getContingencyElements().stream().map(ContingencyElementDTO::toDto).toList(),
+                subjectLimitViolations
+            );
+        }).toList();
+    }
+
+    @Transactional(readOnly = true)
+    public List<SubjectLimitViolationToContingencyDTO> findNmKConstraintsResult(UUID resultUuid) {
+        assertResultExists(resultUuid);
+        List<SubjectLimitViolationEntity> subjectLimitViolations = subjectLimitViolationRepository.findByResultIdOrderBySubjectId(resultUuid);
+
+        return subjectLimitViolations.stream().map(subjectLimitViolation -> {
+            // we only keep converged contingencies here
+            List<ContingencyFromSubjectLimitViolationDTO> contingencies = subjectLimitViolation.getContingencyLimitViolations().stream()
+                .filter(lm -> LoadFlowResult.ComponentResult.Status.CONVERGED.name().equals(lm.getContingency().getStatus()))
+                .map(ContingencyFromSubjectLimitViolationDTO::toDto)
+                .toList();
+
+            return new SubjectLimitViolationToContingencyDTO(subjectLimitViolation.getSubjectId(), contingencies);
+        }).toList();
+    }
+
+    public void assertResultExists(UUID resultUuid) {
+        if (securityAnalysisResultRepository.findById(resultUuid).isEmpty()) {
+            throw new SecurityAnalysisException(SecurityAnalysisException.Type.RESULT_NOT_FOUND);
+        }
+    }
+
+    @Transactional
+    public void insert(UUID resultUuid, SecurityAnalysisResult result, SecurityAnalysisStatus status) {
+        Objects.requireNonNull(resultUuid);
+        Objects.requireNonNull(result);
+
+        SecurityAnalysisResultEntity securityAnalysisResult = SecurityAnalysisResultEntity.toEntity(resultUuid, result, status);
+        securityAnalysisResultRepository.save(securityAnalysisResult);
```

Comment:
Remove `save`

Reply:
I don't think I should since we're creating a new Entity
tests are not passing anymore if I do